doc/tutorials/error_analysis/error_analysis_part1.ipynb

```diff
@@ -0,0 +1,464 @@
+   "metadata": {
+    "solution2": "hidden"
+   },
+   "source": [
+    "```python\n",
+    "def do_binning_analysis(data, bin_size):\n",
+    "    n_samples = len(data)\n",
+    "    n_bins = n_samples // bin_size\n",
+    "    bin_avgs = np.zeros(n_bins)\n",
+    "    for i in range(n_bins):\n",
+    "        bin_avgs[i] = np.average(data[i*bin_size:(i+1)*bin_size])\n",
+    "    avg = np.average(bin_avgs)\n",
+    "    sem = np.sqrt(np.sum((bin_avgs-avg)**2)/((n_bins-1.5)*n_bins))\n",
+    "    return sem\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "solution2": "hidden",
+    "solution2_first": true
+   },
+   "source": [
+    "#### Exercise\n",
+    "Now take the data in ```time_series_1``` and perform binning analysis for bin sizes from 3 up to 5000 and plot the estimated SEMs against the bin size with logarithmic x axis. Your SEM estimates should be stored in a numpy array called ```sems```."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "solution2": "hidden"
+   },
+   "source": [
+    "```python\n",
+    "sizes = np.arange(3,5001,dtype=int)\n",
+    "sems = np.zeros(5001-3,dtype=float)\n",
+    "for s in range(len(sizes)):\n",
+    "    sems[s] = do_binning_analysis(time_series_1,sizes[s])\n",
+    "\n",
+    "plt.plot(sizes,sems, \"x\")\n",
+    "plt.xscale(\"log\")\n",
+    "plt.show()\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You should see that the series converges to a value between 0.02 and 0.03, before transitioning into a noisy tail. The tail becomes increasingly noisy, because as the block size increases, the number of blocks decreases, thus resulting in worse statistics.\n",
+    "\n",
+    "To extract the correct SEM from this plot, we can fit an exponential function to the first part of the data, that doesn't suffer from too much noise."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scipy.optimize import curve_fit\n",
+    "\n",
+    "# only fit to the first couple of SEMs\n",
+    "CUTOFF = 600\n",
+    "\n",
+    "# sizes of the corresponding bins\n",
+    "sizes = np.arange(3,3+CUTOFF,dtype=int)\n",
+    "\n",
+    "def fit_fn(x,a,b,c):\n",
+    "    return -np.exp(-a*x)*b+c\n",
+    "\n",
+    "fit_params, _ = curve_fit(fit_fn,sizes,sems[:CUTOFF], (0.05,1,0.5))\n",
+    "\n",
+    "fit_sems = fit_fn(sizes,*fit_params)\n",
+    "\n",
+    "plt.plot(sizes,sems[:CUTOFF], \"x\")\n",
+    "plt.plot(sizes,fit_sems,\"-\")\n",
+    "plt.xscale(\"log\")\n",
+    "plt.show()\n",
+    "\n",
+    "print(\"Final Standard Error of the Mean: {:.3f}\".format(fit_params[2]))"
```

Comment:
f-strings look more modern: e3258a4ee6ba01b697fdfe9a25c8acf216e424a3

Reply:
You probably also want to look into the `logging` module instead of `print`. We use it in the LB tutorial part 2. Example:
```python
import logging
logging.basicConfig(level=logging.INFO, stream=sys.stdout)
# ...
logging.info("Equilibrating the system.")
```
Output:
```
INFO:root:Equilibrating the system.
```